### PR TITLE
R4R: Fix not find validator set bug

### PR DESCRIPTION
**Pull request by zhiqiang-bianjie**
Resolves : #2045

### CODE REVIEW

File: server/snapshot.go [hunk 1]
```diff
@@ -3,6 +3,7 @@ package server
 import (
 	"errors"
 	"fmt"
+	"github.com/irisnet/irishub/codec"
 	"io"
 	"os"
 	"path/filepath"
```
Comment (chengwenxi) [inline server/snapshot.go]: go import

File: server/reset.go [hunk 2]
```diff
@@ -53,3 +61,21 @@ func ResetCmd(ctx *Context, cdc *codec.Codec, appReset AppReset) *cobra.Command
...
+		if r := recover(); r != nil {
+			panic(errors.Errorf("height: %d not existed in block store", target))
+		}
+
+	}()
+	blockStore := bc.NewBlockStore(blockDb)
+	block := blockStore.LoadBlock(target)
+	if block == nil {
+		return errors.Errorf("height: %d not existed in block store", target)
+	}
```
Comment (chengwenxi) [inline server/reset.go]: seems execute never

Comment (zhiqiang-bianjie) [reply server/reset.go]: block may be null pointer

Comment (zhiqiang-bianjie) [reply server/reset.go]: > seems execute never

block may be null pointer

**Pull request merged**